src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -168,7 +168,8 @@ class Config:
             if os.path.exists(corepath):
                 # FIXME: sysctl module not working
                 await run('sysctl', f"kern.corefile='{corepath}/%N.core'")
-                os.chmod(corepath, 0o775)
+                # 777 permissions are set to make sure that other users can also dump their core
```

Comment:
I am not convinced that is such a good idea. Allowing any user to write there may pose as a risk for e.g. writing tons of data and getting the disk full.

I think at a minimum we should set a quota for that dataset, but that is beyond the scope of the issue for 11.1.

Lets please discard this change for 11.1 and discuss and potentially do more extensive changes for 11.3

Reply:
Makes sense. Updated the PR with latest changes. Thank you